### PR TITLE
Support credentials in deploy package

### PR DESCRIPTION
**Pull request by midigofrank**
## Short Description

Support the management of project credentials in via `project.yaml` - works with the `support-credentials-in-cli-deploy` branch in lightning

## Implementation Details

- [x] When you `pull`, the state is updated with the `project wide credentials`
- [x] You can update the job credential by pointing it to an existing `credential key`  
- [x] You can create a new project_credential if you (the apiToken user) own that credential

## QA Notes

List any considerations/cases/advice for testing/QA here.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added unit tests
- [x] Changesets have been added (if there are production code changes)

## Release branch checklist

Delete this section if this is not a release PR.

If this IS a release branch:

- [x] Run `pnpm changeset` from root to bump versions
- [x] Run `pnpm install`
- [x] Commit the new version numbers
- [x] Run `pnpm changeset tag` to generate tags
- [x] Push tags `git push --tags`

Tags may need updating if commits come in after the tags are first generated.

### CODE REVIEW

**Review by christad92 — APPROVED** [review]

**Comment by taylordowns2000** [comment]
@midigofrank , can you resolve the merge conflicts here?

i've also requested a review from @stuartc as he's likely to have some opinions 😄

File: packages/deploy/src/stateTransform.ts [hunk 1]
```diff
@@ -20,7 +21,25 @@ import {
...
+    specJobCredential &&
+    typeof stateCredentials[specJobCredential] === 'undefined'
+  ) {
+    throw new DeployError(
+      `Could not find a credential with name: ${specJobCredential}`,
+      'VALIDATION_ERROR'
+    );
+  }
+
+  return specJobCredential && stateCredentials[specJobCredential].id;
```
Comment (josephjclark) [inline packages/deploy/src/stateTransform.ts:38]: You've already tested that `specJobCredential` is defined - you can just go ahead and return the value here

File: packages/deploy/src/stateTransform.ts [hunk 2]
```diff
@@ -20,7 +21,25 @@ import {
 import { DeployError } from './deployError';
 import { Logger } from '@openfn/logger/dist';
 
+function getStateJobCredential(
+  specJobCredential: undefined | null | string,
+  stateCredentials: ProjectState['project_credentials']
+): undefined | null | string {
+  if (
```
Comment (josephjclark) [inline packages/deploy/src/stateTransform.ts:28]: I would simply set the condition to
```
if (!stateCredentials[specJobCredential]) {
```

That's more idiomatic, concise, and safely covers what you're looking for

File: packages/deploy/src/stateTransform.ts (hunk 1 above)
Comment (midigofrank) [reply packages/deploy/src/stateTransform.ts:38]: But it can be `null`

**Review by josephjclark — APPROVED** [review]
Haven't tested anything but happy with the code.

When you want to release, can you please follow the Release Process steps in the PR?

**Pull request merged**